black-jack-redo.rb

```diff
@@ -119,67 +227,51 @@ def view_deck(array)
 	end
 
 	input = gets.chomp
-	input.downcase!      
-
-	if input == 'd' && cards == 52
-		#print_with_stars()
-		puts 'Your were dealt: '
-		user.push(array.pop)
-		user.push(array.pop)
-		puts array.length
-		puts user.length
-		#value = handcount(array)
-		#puts "Your cards add up to #{value} points!"
-		cards = 50
-		#format
-		puts '-------'
-		gameloop(array, user, cards)
-	elsif input == 'd' && cards != 52
-		#print_with_stars()
-		count4 = handcount(array)
-		ha, bool = drawone(array, cards, count4)
-		user.push(ha)
-		#user.push(drawone(array, cards, count4))
-		cards = cards - 1
-		#format
-		puts '-------'
-		if bool == true
-			array = []
-			user = []
-			cards = 52
-			array = array_gen(array)
-			gameloop(array, user, cards)
+	input.downcase!    
+	input = input[0]
+
+	if input == 'd' && current_deck.length == full_deck_card_count
+		player_hand_as_string, bust = drawone_card(current_deck, player_hand)
+		player_hand_as_string, bust = drawone_card(current_deck, player_hand)
```

Comment:
Why is this here twice?

Reply:
The lines repeat themselves because the first deal requires two cards to be drawn for both the player and the dealer. I still went back and cleaned it up though by moving that code into a method called deal_initial_hand